### PR TITLE
Add monsters and bosses from 11.02 update

### PR DESCRIPTION
**Pull request by omarcopires**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed 
* All monsters from the update have been added.
* All monster looktypes were taken from staticdata.
* Bosses may have slightly different looktype colors.

**Issues addressed:** 
#4107

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/monster/monsters/ice_dragon.xml [hunk 1]
```diff
@@ -39,11 +53,10 @@
 		<element deathPercent="0" />
 	</elements>
 	<immunities>
+		<immunity fire="1" />
```
Comment (torresmon) [inline data/monster/monsters/ice_dragon.xml:56]:
```suggestion
		<immunity ice="1" />
```

File: data/monster/monsters.xml [hunk 2]
```diff
@@ -27,6 +32,7 @@
 	<monster name="Axeitus Headbanger" file="monsters/axeitus_headbanger.xml" />
 	<monster name="Azerus" file="monsters/azerus.xml" />
 	<monster name="Azure Frog" file="monsters/azure_frog.xml" />
+	<monster name="Baby Dragon" file="monsters/baby_dragon_(creature).xml" />
```
Comment (EPuncker) [inline data/monster/monsters.xml:35]:
```suggestion
	<monster name="Baby Dragon" file="monsters/baby_dragon.xml" />
```

File: data/monster/monsters.xml [hunk 3]
```diff
@@ -352,9 +376,11 @@
 	<monster name="Husky" file="monsters/husky.xml" />
 	<monster name="Hyaena" file="monsters/hyaena.xml" />
 	<monster name="Hydra" file="monsters/hydra.xml" />
+	<monster name="Ice Dragon" file="monsters/ice_dragon.xml" />
 	<monster name="Ice Golem" file="monsters/ice_golem.xml" />
 	<monster name="Ice Overlord" file="monsters/ice_overlord.xml" />
 	<monster name="Ice Witch" file="monsters/ice_witch.xml" />
+	<monster name="Icicle" file="monsters/icicle_(creature).xml" />
```
Comment (EPuncker) [inline data/monster/monsters.xml:383]:
```suggestion
	<monster name="Icicle" file="monsters/icicle.xml" />
```

File: data/monster/monsters/a_shielded_astral_glyph.xml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		<flag canpushitems="0" />
+		<flag canpushcreatures="0" />
+		<flag staticattack="90" />
+		<flag targetdistance="1" />
+		<flag canwalkonenergy="0" />
+		<flag canwalkonfire="0" />
+		<flag canwalkonpoison="0" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" skill="90" attack="100" />
```
Comment (EPuncker) [inline data/monster/monsters/a_shielded_astral_glyph.xml:24]:
```suggestion
		<attack name="melee" interval="2000" min="0" max="-780" />
```

File: data/monster/monsters/twisted_shaper.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="Twisted Shaper" nameDescription="a twisted shaper" experience="1750" speed="280" >
+	<health now="2500" max="2500" />
+	<look type="932" head="105" body="0" legs="0" feet="94" addons="1" mount="0" corpse="0" />
```
Comment (EPuncker) [inline data/monster/monsters/twisted_shaper.xml:4]:
```suggestion
	<look type="932" head="105" body="0" legs="0" feet="94" corpse="27724" />
```

File: data/monster/monsters/twisted_shaper.xml [hunk 6]
```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="Twisted Shaper" nameDescription="a twisted shaper" experience="1750" speed="280" >
```
Comment (EPuncker) [inline data/monster/monsters/twisted_shaper.xml:2]:
```suggestion
<monster name="Twisted Shaper" nameDescription="a twisted shaper" race="blood" experience="1750" speed="280">
```

File: data/monster/monsters/twisted_shaper.xml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+		<flag canpushitems="1" />
+		<flag canpushcreatures="1" />
+		<flag staticattack="90" />
+		<flag targetdistance="1" />
+		<flag canwalkonenergy="1" />
+		<flag canwalkonfire="0" />
+		<flag canwalkonpoison="1" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" chance="100" min="0" max="-100" />
```
Comment (EPuncker) [inline data/monster/monsters/twisted_shaper.xml:24]:
```suggestion
		<attack name="melee" interval="2000" min="0" max="-100" />
```

File: data/monster/monsters/mounted_thorn_knight.xml [hunk 8]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="Mounted Thorn Knight" nameDescription="Mounted Thorn Knight" race="blood" experience="15000" speed="430" manacost="290">
```
Comment (EPuncker) [inline data/monster/monsters/mounted_thorn_knight.xml:2]:
```suggestion
<monster name="Mounted Thorn Knight" nameDescription="Mounted Thorn Knight" race="blood" experience="15000" speed="430">
```

File: data/monster/monsters/mounted_thorn_knight.xml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+		<flag canpushitems="0" />
+		<flag canpushcreatures="0" />
+		<flag staticattack="90" />
+		<flag targetdistance="1" />
+		<flag canwalkonenergy="0" />
+		<flag canwalkonfire="0" />
+		<flag canwalkonpoison="0" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" chance="100" min="0" max="-1032" />
```
Comment (EPuncker) [inline data/monster/monsters/mounted_thorn_knight.xml:24]:
```suggestion
		<attack name="melee" interval="2000" min="0" max="-1032" />
```

File: data/monster/monsters/orclops_ravager.xml [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+		<flag canpushitems="1" />
+		<flag canpushcreatures="1" />
+		<flag staticattack="90" />
+		<flag targetdistance="1" />
+		<flag canwalkonenergy="0" />
+		<flag canwalkonfire="0" />
+		<flag canwalkonpoison="1" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" chance="100" min="0" max="-132" />
```
Comment (EPuncker) [inline data/monster/monsters/orclops_ravager.xml:24]:
```suggestion
		<attack name="melee" interval="2000" min="0" max="-132" />
```

**Review by EPuncker — CHANGES_REQUESTED** [review]
first review

File: data/monster/monsters/broken_shaper.xml [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+		<element earthPercent="20" />
+		<element firePercent="-15" />
+	</elements>
+	<immunities>
+		<immunity paralyze="1" />
+		<immunity invisible="1" />
+		<immunity drunk="1" />
+	</immunities>
+	<voices interval="5000" chance="10">
+		<voice sentence="&lt;grunt&gt;" yell="false" />
```
[the remark this answers is not in the document]
Comment (omarcopires) [reply data/monster/monsters/broken_shaper.xml]: image

[the remark this answers is not in the document]
Comment (omarcopires) [reply data/monster/monsters/broken_shaper.xml]: image

File: data/monster/monsters/icicle.xml [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+		<flag targetdistance="1" />
+		<flag canwalkonenergy="0" />
+		<flag canwalkonfire="0" />
+		<flag canwalkonpoison="0" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" min="0" max="-219" />
+	</attacks>
+	<defenses armor="10" defense="5">
+		<defense name="icecle heal" interval="2000" chance="25" />
```
Comment (EPuncker) [inline data/monster/monsters/icicle.xml:27]: icicle* same with the lua file

File: data/monster/monsters/kalyassa.xml [hunk 13]
```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="Kalyassa" nameDescription="kalyassa" race="blood" experience="9000" speed="350">
```
Comment (EPuncker) [inline data/monster/monsters/kalyassa.xml:2]:
```suggestion
<monster name="Kalyassa" nameDescription="Kalyassa" race="blood" experience="9000" speed="350">
```

File: data/monster/monsters/kalyassa.xml [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+			<attribute key="shootEffect" value="fire" />
+		</attack>
+	</attacks>
+	<defenses armor="10" defense="5">
+		<defense name="healing" interval="2000" chance="20" min="600" max="700">
+			<attribute key="areaEffect" value="blueshimmer" />
+		</defense>
+	</defenses>
+	<summons maxSummons="2">
+		<summon name="Dragon Lord" interval="2000" chance="20" max="2" />
```
Comment (EPuncker) [inline data/monster/monsters/kalyassa.xml:45]: already defined at line 44
```suggestion
		<summon name="Dragon Lord" interval="2000" chance="20" />
```

File: data/monster/monsters/lost_time.xml [hunk 15]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="Lost Time" nameDescription="a lost time" race="undead" experience="0" speed="160">
+	<health now="2000" max="2000" />
+	<look typeex="0" corpse="0" />
```
Comment (EPuncker) [inline data/monster/monsters/lost_time.xml:4]:
```suggestion
	<look typeex="26385" corpse="0" />
```

File: data/monster/monsters/melting_frozen_horror.xml [hunk 16]
```diff
@@ -0,0 +1,53 @@
...
+			<attribute key="shootEffect" value="ice" />
+			<attribute key="areaEffect" value="yellowspark" />
+		</attack>
+		<attack name="speed" interval="2000" chance="11" length="8" spread="0" speedchange="-800" duration="20000">
+			<attribute key="areaEffect" value="energy" />
+		</attack>
+		<attack name="hirintror skill reducer" interval="2000" chance="10" />
+	</attacks>
+	<defenses armor="10" defense="5">
+		<defense name="healing" interval="2000" chance="20" radius="7" min="0" max="0">
```
Comment (EPuncker) [inline data/monster/monsters/melting_frozen_horror.xml:36]: remove the healing or add values

File: data/monster/monsters/solid_frozen_horror.xml [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+			<attribute key="shootEffect" value="ice" />
+			<attribute key="areaEffect" value="yellowspark" />
+		</attack>
+		<attack name="speed" interval="1000" chance="11" length="8" spread="0" speedchange="-800" duration="20000">
+			<attribute key="areaEffect" value="energy" />
+		</attack>
+		<attack name="hirintror skill reducer" interval="2000" chance="10" />
+	</attacks>
+	<defenses armor="60" defense="60">
+		<defense name="healing" interval="2000" chance="20" radius="7" min="0" max="0">
```
Comment (EPuncker) [inline data/monster/monsters/solid_frozen_horror.xml:36]: remove healing or add values

File: data/monster/monsters/the_last_lore_keeper.xml [hunk 18]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="The Last Lore Keeper" nameDescription="The Last Lore Keeper" race="undead" experience="45000" speed="450">
+	<health now="750000" max="750000" />
+	<look type="939" corpse="0" />
```
Comment (EPuncker) [inline data/monster/monsters/the_last_lore_keeper.xml:4]: missing corpse since it have loot

File: data/monster/monsters/thorn_minion.xml [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+		<flag canpushitems="0" />
+		<flag canpushcreatures="0" />
+		<flag staticattack="90" />
+		<flag targetdistance="1" />
+		<flag canwalkonenergy="0" />
+		<flag canwalkonfire="0" />
+		<flag canwalkonpoison="0" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" min="-60" max="-150" poison="500" />
```
Comment (EPuncker) [inline data/monster/monsters/thorn_minion.xml:24]:
```suggestion
		<attack name="melee" interval="2000" min="0" max="-150" poison="500" />
```

File: data/spells/scripts/monster/frozen_minion_beam.lua [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_POFF)
+combat:setParameter(COMBAT_PARAM_AGGRESSIVE, 0)
+combat:setArea(createCombatArea(AREA_BEAM7))
+
+function onTargetCreature(creature, target)
+	local min = 200
+	local max = 700
+	local master = target:getMaster()
+	if target:isPlayer() and not master or master and master:isPlayer() then
+		doTargetCombatHealth(0, target, COMBAT_ICEDAMAGE, min, max, CONST_ME_NONE)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_beam.lua:11]:
```suggestion
		doTargetCombat(0, target, COMBAT_ICEDAMAGE, min, max, CONST_ME_NONE)
```

File: data/spells/scripts/monster/frozen_minion_beam.lua [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+		return true
+	end
+
+	doTargetCombatHealth(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_beam.lua:15]:
```suggestion
	doTargetCombat(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```

File: data/spells/scripts/monster/frozen_minion_beam.lua [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+	return true
+end
+
+combat:setCallback(CALLBACK_PARAM_TARGETCREATURE, "onTargetCreature")
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_beam.lua:22]:
```suggestion
	return combat:execute(creature, variant)
```

File: data/spells/scripts/monster/frozen_minion_heal.lua [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+
+function onTargetCreature(creature, target)
+	local min = 100
+	local max = 200
+	local master = target:getMaster()
+	if target:isPlayer() and not master or master and master:isPlayer() then
+		return true
+	end
+
+	doTargetCombatHealth(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_heal.lua:14]:
```suggestion
	doTargetCombat(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```

File: data/spells/scripts/monster/frozen_minion_wave.lua [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+	end
+
+	doTargetCombatHealth(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
+	return true
+end
+
+combat:setCallback(CALLBACK_PARAM_TARGETCREATURE, "onTargetCreature")
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_wave.lua:30]:
```suggestion
	return combat:execute(creature, variant)
```

File: data/spells/scripts/monster/frozen_minion_wave.lua [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_POFF)
+combat:setParameter(COMBAT_PARAM_AGGRESSIVE, 0)
+combat:setArea(createCombatArea(area))
+
+function onTargetCreature(creature, target)
+	local min = 200
+	local max = 700
+	local master = target:getMaster()
+	if target:isPlayer() and not master or master and master:isPlayer() then
+		doTargetCombatHealth(0, target, COMBAT_ICEDAMAGE, min, max, CONST_ME_NONE)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_wave.lua:19]:
```suggestion
		doTargetCombat(0, target, COMBAT_ICEDAMAGE, min, max, CONST_ME_NONE)
```

File: data/spells/scripts/monster/frozen_minion_wave.lua [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+		return true
+	end
+
+	doTargetCombatHealth(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```
Comment (EPuncker) [inline data/spells/scripts/monster/frozen_minion_wave.lua:23]:
```suggestion
	doTargetCombat(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```

File: data/spells/scripts/monster/hirintror_freeze.lua [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+	{0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0}
+}
+
+local condition = Condition(CONDITION_FREEZING)
+condition:setParameter(CONDITION_PARAM_DELAYED, 1)
+condition:addDamage(25, 8000, -8)
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_ICETORNADO)
+combat:addCondition(condition)
```
Comment (EPuncker) [inline data/spells/scripts/monster/hirintror_freeze.lua:23]: remove this one, duplicated

File: data/spells/scripts/monster/hirintror_freeze.lua [hunk 28]
```diff
@@ -0,0 +1,29 @@
...
+combat:setArea(createCombatArea(area))
+combat:addCondition(condition)
+
+function onCastSpell(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/hirintror_freeze.lua:27]:
```suggestion
function onCastSpell(creature, variant)
```

File: data/spells/scripts/monster/hirintror_freeze.lua [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+	return combat:execute(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/hirintror_freeze.lua:28]:
```suggestion
	return combat:execute(creature, variant)
```

File: data/spells/scripts/monster/hirintror_skill_reducer.lua [hunk 30]
```diff
@@ -0,0 +1,21 @@
...
+	condition:setParameter(CONDITION_PARAM_SKILL_FISTPERCENT, i)
+	condition:setParameter(CONDITION_PARAM_SKILL_DISTANCEPERCENT, i)
+	condition:setParameter(CONDITION_PARAM_SKILL_SHIELDPERCENT, i)
+
+	local area = createCombatArea(AREA_SQUARE1X1)
+	combat[i]:setArea(area)
+	combat[i]:addCondition(condition)
+end
+
+function onCastSpell(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/hirintror_skill_reducer.lua:19]:
```suggestion
function onCastSpell(creature, variant)
```

File: data/spells/scripts/monster/hirintror_skill_reducer.lua [hunk 31]
```diff
@@ -0,0 +1,21 @@
...
+	return combat[math.random(20, 40)]:execute(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/hirintror_skill_reducer.lua:20]:
```suggestion
	return combat[math.random(20, 40)]:execute(creature, variant)
```

File: data/spells/scripts/monster/icecle_heal.lua [hunk 32]
```diff
@@ -0,0 +1,22 @@
...
+
+function onTargetCreature(creature, target)
+	local min = 400
+	local max = 600
+	local master = target:getMaster()
+	if target:isPlayer() and not master or master and master:isPlayer() then
+		return true
+	end
+
+	doTargetCombatHealth(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```
Comment (EPuncker) [inline data/spells/scripts/monster/icecle_heal.lua:14]:
```suggestion
	doTargetCombat(0, target, COMBAT_HEALING, min, max, CONST_ME_NONE)
```

File: data/spells/scripts/monster/lloyd_wave_1.lua [hunk 33]
```diff
@@ -0,0 +1,18 @@
...
+	{0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0},
+	{0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0},
+	{0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 2, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_ENERGYDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, 179)
```
Comment (EPuncker) [inline data/spells/scripts/monster/lloyd_wave_1.lua:13]:
```suggestion
combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_PURPLECHAIN)
```

File: data/spells/scripts/monster/lloyd_wave_1.lua [hunk 34]
```diff
@@ -0,0 +1,18 @@
...
+combat:setArea(createCombatArea(area))
+
+function onCastSpell(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/lloyd_wave_1.lua:16]:
```suggestion
function onCastSpell(creature, variant)
```

File: data/spells/scripts/monster/lloyd_wave_1.lua [hunk 35]
```diff
@@ -0,0 +1,18 @@
...
+	return combat:execute(creature, var)
```
Comment (EPuncker) [inline data/spells/scripts/monster/lloyd_wave_1.lua:17]:
```suggestion
	return combat:execute(creature, variant)
```

File: data/spells/scripts/monster/lloyd_wave_2.lua [hunk 36]
```diff
@@ -0,0 +1,18 @@
...
+	{0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 2, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_ENERGYDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, 179)
```
Comment (EPuncker) [inline data/spells/scripts/monster/lloyd_wave_2.lua:13]: same changes from wave 1

File: data/spells/scripts/monster/lloyd_wave_3.lua [hunk 37]
```diff
@@ -0,0 +1,18 @@
...
+	{0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 2, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_ENERGYDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, 179)
```
Comment (EPuncker) [inline data/spells/scripts/monster/lloyd_wave_3.lua:13]: same changes from wave 1

File: data/spells/scripts/monster/souleater_wave.lua [hunk 38]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_LIFEDRAINDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_SOUND_RED)
+combat:setArea(createCombatArea(area))
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```
Comment (EPuncker) [inline data/spells/scripts/monster/souleater_wave.lua:15]: this spell is not needed, souleater already have this spell coded on his xml file

```
		<attack name="death" interval="2000" chance="10" length="4" spread="3" target="0" min="-100" max="-200">
			<attribute key="areaEffect" value="rednote" />
		</attack>
```

File: data/spells/scripts/monster/white_shade_paralyze.lua [hunk 39]
```diff
@@ -0,0 +1,12 @@
...
+condition:setFormula(-0.3, 0, -0.45, 0)
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_HITAREA)
+combat:setArea(createCombatArea(AREA_CIRCLE2X2))
+combat:addCondition(condition)
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```
Comment (EPuncker) [inline data/spells/scripts/monster/white_shade_paralyze.lua:12]: this spell is not needed in lua, you can use this in xml:

```
		<attack name="speed" interval="2000" chance="15" target="1" radius="2" speedchange="-450" duration="20000">
			<attribute key="areaEffect" value="blackspark" />
		</attack>
```
(don't forget to change spaces to tab when copying from github

File: data/spells/scripts/monster/wyrm_wave.lua [hunk 40]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_ENERGYDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_PURPLEENERGY)
+combat:setArea(createCombatArea(area))
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```
Comment (EPuncker) [inline data/spells/scripts/monster/wyrm_wave.lua:17]: this spell not needed in lua, just use the one that is already in wyrm file and adjust the damage:

```
		<attack name="energy" interval="2000" chance="10" length="5" spread="2" min="-130" max="-200">
			<attribute key="areaEffect" value="purpleenergy" />
		</attack>
```

File: data/spells/scripts/monster/ghastly_dragon_paralyze.lua [hunk 41]
```diff
@@ -0,0 +1,11 @@
...
+condition:setParameter(CONDITION_PARAM_TICKS, 20000)
+condition:setFormula(-0.8, 0, -0.9, 0)
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_BATS)
+combat:addCondition(condition)
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```
Comment (EPuncker) [inline data/spells/scripts/monster/ghastly_dragon_paralyze.lua:11]: lua not needed, use this in xml:

```
		<attack name="speed" interval="2000" chance="20" range="7" target="1" speedchange="-800" duration="20000">
			<attribute key="areaEffect" value="smallclouds" />
		</attack>
```

File: data/spells/scripts/monster/ghastly_dragon_wave.lua [hunk 42]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_DEATHDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_LOSEENERGY)
+combat:setArea(createCombatArea(area))
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```
Comment (EPuncker) [inline data/spells/scripts/monster/ghastly_dragon_wave.lua:16]: lua not needed, use this in xml (adjust damage if needed):

```
		<attack name="death" interval="2000" chance="10" length="8" spread="3" min="-120" max="-250">
			<attribute key="areaEffect" value="bluebubble" />
		</attack>
```

**Review by EPuncker — CHANGES_REQUESTED** [review]

File: data/monster/monsters/the_last_lore_keeper.xml (hunk 18 above)
Comment (omarcopires) [reply data/monster/monsters/the_last_lore_keeper.xml:4]: This boss has no corpse, loot goes directly to reward bag

Comment (EPuncker) [inline data/monster/monsters/the_last_lore_keeper.xml:4]:
```suggestion
	<look type="27741" corpse="0" />
```

File: data/monster/monsters/the_last_lore_keeper.xml [hunk 43]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<monster name="The Last Lore Keeper" nameDescription="The Last Lore Keeper" race="undead" experience="45000" speed="450">
+	<health now="750000" max="750000" />
+	<look type="27741" corpse="0" />
```
Comment (EPuncker) [inline data/monster/monsters/the_last_lore_keeper.xml:4]:
```suggestion
	<look type="27741" corpse="27741" />
```

File: data/monster/monsters/lloyd.xml [hunk 44]
```diff
@@ -0,0 +1,86 @@
...
+		<flag canwalkonpoison="0" />
+	</flags>
+	<attacks>
+		<attack name="melee" interval="2000" min="0" max="-1200" />
+		<attack name="energy" interval="2000" chance="12" length="6" spread="0" min="-130" max="-460">
+			<attribute key="areaEffect" value="purpleenergy" />
+		</attack>
+		<attack name="lloyd wave 1" interval="2000" chance="12" min="-430" max="-560" />
+		<attack name="lloyd wave 2" interval="2000" chance="12" min="-230" max="-460" />
+		<attack name="lloyd wave 3" interval="2000" chance="12" min="-430" max="-660" />
```
Comment (EPuncker) [inline data/monster/monsters/lloyd.xml:30]:
```suggestion
		<attack name="energy" interval="2000" chance="15" radius="2" target="0" min="-430" max="-660">
			<attribute key="areaEffect" value="purplechain" />
		</attack>
```
and delete the lua file

**Review by EPuncker — APPROVED** [review]

**Comment by soul4soul** [comment]
There are spells named `hirintror xyz` being used on a monster named "Melting Frozen Horror". "Hirintror" and "Melting Frozen Horror" are different monsters. TibiaWiki has limited information on both creatures so I can't tell which one should have freezing damage,

**Comment by torresmon** [comment]
> There are spells named `hirintror xyz` being used on a monster named "Melting Frozen Horror". "Hirintror" and "Melting Frozen Horror" are different monsters. TibiaWiki has limited information on both creatures so I can't tell which one should have freezing damage,

Yeah those are a pain. I think they all set the freezing condition. I'll check some videos and try to get more information for you on this

**Pull request merged**